backend/src/supervisors/supervisor.py

```diff
@@ -12,20 +12,25 @@
 no_agent_response = "I am sorry, but I was unable to find an agent to solve this task"
 
 
+async def process_question(question) -> None:
+    try:
+        (agent_name, answer, status) = await solve_task(question, get_scratchpad())
+        update_scratchpad(agent_name, question, answer)
+        if status == "error":
+            raise Exception(answer)
+    except Exception as error:
+        update_scratchpad(error=str(error))
+
+
 async def solve_all(intent_json) -> None:
     questions = intent_json["questions"]
```

Comment:
```suggestion
        questions = []
        if len(intent_json["questions"]) > 0:
            questions += intent_json["questions"]
        else:
            questions.append(intent_json["question"])
```
and then you just need to loop over questions. Might make this a bit more clear?